### PR TITLE
Parse requirements from file

### PR DESCRIPTION
**Pull request by JamesOwers**
Reads minimal requirements from the adjacent requirements.txt and uses them to define the install_requires
options for setuptools.

This addresses https://github.com/cuthbertLab/music21/issues/338 and fixes https://github.com/cuthbertLab/music21/issues/1050.

I tested using the following procedure:
```
git_home=~/Downloads
repo_name=music21
repo_path=${git_home}/${repo_name}
repo_url=https://github.com/JamesOwers/music21.git
git clone ${repo_url} ${repo_path}
conda create -n ${repo_name}-dev python=3.8 --yes
conda activate ${repo_name}-dev
pip install ${repo_path}
python -c 'import jsonpickle' && echo 'This succeeded as expected!'
```

### CODE REVIEW

**Comment by jacobtylerwalls** [comment]
@mscuthbert would you prefer hard-coding here? I don't know how common it is to read files in a setup.py

**Comment by jacobtylerwalls** [comment]
Ah, the python docs example from pip's setup.py does read files.

**Comment by JamesOwers** [comment]
Yeah, tis standard (check out numpy/pandas/sklearn etc). If you don’t use
the requirements.txt elsewhere, it’s somewhat neater to put these options
in the setup.cfg IMHO - removes two files and the file reading code in
setup.py. Happy to implement that if you prefer.

**Review by mscuthbert — COMMENTED** [review]
I wonder if at this point, with numpy and matplotlib so commonly available if it's just worth it to put them and jsonpickle into the requirements.txt file -- I used to want to keep the install super minimal for people on embedded systems, but now I wonder how many people are running a Python installation that can't really run them.

I wish that there were a robust solution to the problem I posted at https://stackoverflow.com/questions/53376135/install-by-default-optional-dependencies-in-python-setuptools

File: setup.py [hunk 1]
```diff
@@ -31,12 +31,17 @@
...
-]
+min_requirements_path = os.path.join(
+    os.path.dirname(__file__), "requirements-minimum.txt"
+)
+with open(min_requirements_path, 'r') as f:
+    INSTALL_REQUIRES = f.read().splitlines()
+all_requirements_path = os.path.join(os.path.dirname(__file__), "requirements.txt")
+with open(all_requirements_path, 'r') as f:
+    extra_requirements = [
+        req for req in f.read().splitlines() if req not in INSTALL_REQUIRES
```
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply setup.py:42]: They should both specify a minimum version (my bad!)

**Comment by jacobtylerwalls** [comment]
> I wonder if at this point, with numpy and matplotlib so commonly available if it's just worth it to put them and jsonpickle into the requirements.txt file

+1. Users of tiny systems can download zip and `python -m pip install -r requirements-minimum.txt` if they don't want matplotlib and friends, no?

**Comment by mscuthbert** [comment]
Okay, let's go with that.  The extras are then just scipy, pygame, and perhaps even more obscure single-use calls.  I've removed Pillow and Gregorio at some point.

In a certain sense, MuseScore and Lilypond are more central deps but they're not python modules so can't be installed by PIP.

**Comment by mscuthbert** [comment]
By go-with-that, I mean, we'll just put numpy, matplotlib, and jsonpickle in requirements.txt.

File: setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply setup.py:42]: Specifying the minimum version in both places was done in #1055

**Comment by JamesOwers** [comment]
Have removed the extras_require option - the install now reads the requirements file, and pulled from main (which adds extra packages to requirements.txt).

**Review by jacobtylerwalls — APPROVED** [review]

**Pull request merged**